### PR TITLE
Middleware for Italy

### PR DESCRIPTION
**Pull request by TSchmiedlechner**
A first draft of the Middleware interface documentation for Italy.

- [x] Introduction
- [x] Reference tables
- [x] Terminology
- [x] Installation
- [ ] ...

### CODE REVIEW

File: doc/middleware-it/reference-tables/type-of-service-ftchargeitemcase.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+| `0x4954000000000027` | **Discount/Surcharge negativ value is discount, positiv value surcharge: zero Vat  0%**<br /> | TBD |
+
+## ftChargeItemCaseFlag
+This table shows flags that can be added to each `ftChargeItemCase` with values applicable to the Italian market. 
+
+| **Value**            | **Description**                                                                                        | **Middleware Version** |
+| -------------------- | -------------- | ---------------------- |
+| `0x4954000000010000` | On void receipt, Operationtype: Acconto | TBD |
+| `0x4954000000020000` | On void receipt, Operationtype: Free Of Charge | TBD |
+| `0x4954000000030000` | On void receipt, Operationtype: Single Use Voucher | TBD |
```
Comment (TSchmiedlechner) [inline doc/middleware-it/reference-tables/type-of-service-ftchargeitemcase.md:61]: As we use binary operators on the cases, this must be `0x4954000000040000`

**Review by Bluette-C-Riviere — DISMISSED** [review]

File: doc/middleware-it-registratore-telematico/appendix-it-registratore-telematico.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+slug: /poscreators/middleware-doc/italy
+title: Introduction
+---
+
+This appendix expands on the General Part's information by adding details specific to the Italian market. This additional information is provided only where applicable. The remaining chapters, for which there is no further information required, were omitted.
+
+<div class="alert alert--warning" role="alert">Please note that this information is only complete when combined with the <b>General</b> part. To implement the Middleware, users should get themselves familiar with the general information first and then refer to the country-specific details listed here.</div>
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/appendix-it-registratore-telematico.md:8]:
```suggestion
:::caution

Please note that this information is only complete when combined with the **General** part. To implement the Middleware, users should get themselves familiar with the general information first and then refer to the country-specific details listed here.

:::
```

[I think thats the same thing right?](https://docusaurus.io/docs/markdown-features/admonitions)

File: doc/middleware-it-registratore-telematico/cash-register-integration/cash-register-integration.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+slug: /poscreators/middleware-doc/italy/cash-register-integration
+title: Cash register integration
+---
+
+# Cash register integration
+
+This chapter describes the cash register integration in accordance with Italian law. The general rules for cash register integration are described in the General Part of this document.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/cash-register-integration/cash-register-integration.md:8]:
```suggestion
This chapter describes the cash register integration in accordance with italian law. The general rules for cash register integration are described in the General Part of this document.
```

File: doc/middleware-it-registratore-telematico/communication/communication.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+---
+slug: /poscreators/middleware-doc/italy/communication
+title: Communication
+---
+
+## Communication
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/communication/communication.md:6]:
```suggestion
# Communication
```

File: doc/middleware-it-registratore-telematico/communication/communication.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+This chapter expands more on describing the ways to communicate with the fiskaltrust.Middleware covered in the [Communication](../../general/communication/communication.md) chapter of the General Part, with country-specific information applicable to the Italian market.
+
+### Supported protocols
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/communication/communication.md:9]:
```suggestion
## Supported protocols
```

File: doc/middleware-it-registratore-telematico/communication/communication.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+The Italian Middleware natively supports REST, gRPC and WCF communication without the requirement to use any helper packages. gRPC communication is described in the following section, please refer to the [general part](../../general/communication/communication.md) for more details about REST and WCF.
+
+#### gRPC
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/communication/communication.md:12]:
```suggestion
### gRPC
```

File: doc/middleware-it-registratore-telematico/communication/communication.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+---
+slug: /poscreators/middleware-doc/italy/communication
+title: Communication
+---
+
+## Communication
+This chapter expands more on describing the ways to communicate with the fiskaltrust.Middleware covered in the [Communication](../../general/communication/communication.md) chapter of the General Part, with country-specific information applicable to the Italian market.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/communication/communication.md:7]:
```suggestion
This chapter expands more on describing the ways to communicate with the fiskaltrust.Middleware covered in the [Communication](../../general/communication/communication.md) chapter of the General Part, with country-specific information applicable to the italian market.
```

File: doc/middleware-it-registratore-telematico/communication/communication.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+### Supported protocols
+The Italian Middleware natively supports REST, gRPC and WCF communication without the requirement to use any helper packages. gRPC communication is described in the following section, please refer to the [general part](../../general/communication/communication.md) for more details about REST and WCF.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/communication/communication.md:10]:
```suggestion
The italian Middleware natively supports REST, gRPC and WCF communication without the requirement to use any helper packages. gRPC communication is described in the following section, please refer to the [general part](../../general/communication/communication.md) for more details about REST and WCF.
```

File: doc/middleware-it-registratore-telematico/data-structures/data-structures.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+---
+slug: /poscreators/middleware-doc/italy/data-structures
+title: Data structures
+---
+
+## Data structures
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/data-structures/data-structures.md:6]:
```suggestion
# Data structures
```

File: doc/middleware-it-registratore-telematico/data-structures/data-structures.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+This chapter expands on the descriptions of the country-specific Data Structures, covered in the Chapter ["Data Structures"](../../general/data-structures/data-structures.md) of the General Part, with information applicable to the Italian market.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/data-structures/data-structures.md:8]:
```suggestion
This chapter expands on the descriptions of the country-specific Data Structures, covered in the Chapter ["Data Structures"](../../general/data-structures/data-structures.md) of the General Part, with information applicable to the italian market.
```

File: doc/middleware-it-registratore-telematico/data-structures/data-structures.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+### Receipt Request
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/data-structures/data-structures.md:10]:
```suggestion
## Receipt Request
```

File: doc/middleware-it-registratore-telematico/data-structures/data-structures.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+#### Single fields
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/data-structures/data-structures.md:12]:
```suggestion
### Single fields
```

File: doc/middleware-it-registratore-telematico/data-structures/data-structures.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+| `cbUser`                     | `string (50)`        | Mandatory                                             | Name (not ID) of the user who creates the receipt.                                                                                                                                                                                                                                                                     | 1.3         |
+| `cbReceiptReference`         | `string (50)`        | Mandatory | Used to connect all requests referring to the same business action.                                                                                                                                                                                                                                                    | 1.3         |
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/data-structures/data-structures.md:20]: The `cbReceiptReference` has to be unique in italy. `cbPreviousReceiptReference` is used for refunds.

IDK if theres a need to connect receipts like that in italy... If yes we could also use the `cbPreviousReceiptReference` for that.

File: doc/middleware-it-registratore-telematico/installation/installation.md [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+
+This chapter expands on the installation documentation covered in [Installation in General Part](../../general/installation/installation.md), with country-specific information applicable to the Italian market. 
+
+## fiskaltrust.Middleware
+
+### Windows, Linux, Mac
+
+The mentioned platforms use the fiskaltrust Launcher 2.0 to execute the fiskaltrust.SecurityMechanism. As the Version 2.0 of the Launcher is self-contained, it has native support for all mentioned platforms, which means there are no external dependencies in terms of third party software which needs to be installed.
+
+The following call parameters are available with the launcher `fiskaltrust.exe` for the Italian market (fiskaltrust.Middleware 1.3. and higher):
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/installation/installation.md:16]: The launcher 2.0 exe is `fiskaltrust.Launcher.exe` 😅 also the parameter below is a 1.3 parameter...

maybe we just link to the [launcher docs](https://github.com/fiskaltrust/middleware-launcher#getting-started) here?

File: doc/middleware-it-registratore-telematico/operation-modes/on-premise-installation.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+For hardware requirements, please refer to the [general part](../../general/operation-modes/operation-modes.md).
+
+## Local data storage/platform support matrix
+
+Following local data storage options can be configured in the queue:
+
+|                                                            | Windows       | Linux, macOS  |
+| ---------------------------------------------------------- | ------------- | ------------- |
+| **[SQLite-storage](on-premise-databases/sqlite.md)**       | **supported** | **supported** |
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/operation-modes/on-premise-installation.md:42]: All queues will be supported out of the box in italy 🤔

File: doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/linux.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+| Debian        | >= 10     | 64bit           |
+| Ubuntu        | >= 16.04  | 64bit           |
+
+## Supported packages
+
+### Local data storage/Linux distribution support matrix
+
+|                                                                 | Ubuntu     | Debian     | Other distributions              |
+|-----------------------------------------------------------------|------------|------------|----------------------------------|
+| local [SQLite-Storage](../on-premise-databases/sqlite.md)       | **tested** | **tested** | should work (not tested)         |
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/linux.md:25]: All queues will be supported out of the box in italy 🤔

File: doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/linux.md [hunk 17]
```diff
@@ -0,0 +1,30 @@
+---
+slug: /poscreators/middleware-doc/italy/platforms/linux
+title: Linux
+---
+
+# fiskaltrust.Middleware for Linux and macOs
+
+Starting with version 1.3.45, it's possible to run the Italian Middleware on Linux. Just configure a CashBox and download the launcher via the respective button in the CashBox overview. Like on Windows, the downloaded zip file contains scripts to install and test the Middleware. Other than that, no specific software needs to be installed as the Italian Middleware is running on the Launcher 2.0 which is self-contained and has no external dependencies.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/linux.md:8]:
```suggestion
Starting with version 1.3.45, it's possible to run the italian Middleware on Linux. Just configure a CashBox and download the Launcher via the respective button in the CashBox overview. Like on Windows, the downloaded zip file contains scripts to install and test the Middleware. Other than that, no specific software needs to be installed as the italian Middleware is running on the Launcher 2.0 which is self-contained and has no external dependencies.
```

File: doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/linux.md [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+### Local data storage/Linux distribution support matrix
+
+|                                                                 | Ubuntu     | Debian     | Other distributions              |
+|-----------------------------------------------------------------|------------|------------|----------------------------------|
+| local [SQLite-Storage](../on-premise-databases/sqlite.md)       | **tested** | **tested** | should work (not tested)         |
+
+
+SQLite queues can be launched on Linux, starting from version 1.3.45. 
+
+If you haven't already decided for a communication technology, we strongly recommend gRPC, especially in Linux scenarios, as it provides a cleaner, more stable interface. Please refer to our demo project (currently available in [C#](https://github.com/fiskaltrust/middleware-demo-dotnet), [Java](https://github.com/fiskaltrust/middleware-demo-java), [Node.js](https://github.com/fiskaltrust/middleware-demo-node) and [others](https://github.com/fiskaltrust)) for sample implementations.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/linux.md:30]:
```suggestion
If you haven't already decided for a communication technology, we strongly recommend gRPC, as it provides a cleaner, more stable interface. Please refer to our demo project (currently available in [C#](https://github.com/fiskaltrust/middleware-demo-dotnet), [Java](https://github.com/fiskaltrust/middleware-demo-java), [Node.js](https://github.com/fiskaltrust/middleware-demo-node) and [others](https://github.com/fiskaltrust)) for sample implementations.
```

File: doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/windows.md [hunk 19]
```diff
@@ -0,0 +1,27 @@
+---
+slug: /poscreators/middleware-doc/italy/platforms/windows
+title: Windows
+---
+
+# fiskaltrust.Middleware für Windows
+
+The fiskaltrust.Middleware runs as a service on all 32-bit and 64-bit platforms starting from Windows 7. Older operating systems are not supported as they do not meet the legal requirements of a "safe system environment".
+
+| Operating system                    | Support       |
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/windows.md:10]: Maybe mention that we follow the .net support lifecycle (we support the same OSs as the current LTS .NET versions) and link to the pages?

Comment (volllly) [reply doc/middleware-it-registratore-telematico/operation-modes/on-premise-platforms/windows.md:10]: https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-and-net-core

File: doc/middleware-it-registratore-telematico/operation-modes/scu/epson.md [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+
+### Parameters
+
+| Name | Description | Optional |
+| ---- | ------------ |--------- |
+| DeviceUrl| The URL or IP address of the RT Printer or Server | mandatory |
+| ClientTimeoutMs | The HTTP client timeout used when communicating with the RT Printer or Server. | `15000`<br />optional |
+| ServerTimeoutMs | The server/printer timeout for executing commands.                             |optional               |
+
+Please pay attention to the case-sensitive use of the parameters.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/operation-modes/scu/epson.md:26]: I'm not sure they are case sensitive but I would keep it in none the less ^^

File: doc/middleware-it-registratore-telematico/reference-tables/reference-tables.md [hunk 21]
```diff
@@ -0,0 +1,7 @@
+---
+slug: /poscreators/middleware-doc/italy/reference-tables
+title: Reference tables
+---
+
+# Reference tables
+This chapter expands on the reference tables covered in [Reference Tables in General Part](../../general/reference-tables/reference-tables.md#reference-tables), with country-specific information applicable to the Italian market. The respective tables can be found in the following sub-sections.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/reference-tables/reference-tables.md:7]:
```suggestion
This chapter expands on the reference tables covered in [Reference Tables in General Part](../../general/reference-tables/reference-tables.md#reference-tables), with country-specific information applicable to the italian market. The respective tables can be found in the following sub-sections.
```

File: doc/middleware-it-registratore-telematico/reference-tables/service-status-ftstate.md [hunk 22]
```diff
@@ -0,0 +1,14 @@
+---
+slug: /poscreators/middleware-doc/italy/reference-tables/ftstate
+title: 'Service Status: ftState'
+---
+
+# Service Status: ftState
+
+The country-specific code is made of the country's code value following the ISO-3166-1-ALPHA-2 standard, converted from ASCII into hex. For Italy (IT) this is `0x4954`, which results in `0x4954000000000000` as the value for the "ready" status.
+
+The table below describes its supported statuses for the ftState field following Italian law. These codes can be added through the logic operator `OR`<span id="t-service-status-ftstate-22">.</span>
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/reference-tables/service-status-ftstate.md:10]:
```suggestion
The table below describes its supported statuses for the ftState field following italian law. These codes can be added through the logic operator `OR`<span id="t-service-status-ftstate-22">.</span>
```

File: doc/middleware-it-registratore-telematico/reference-tables/type-of-payment-ftpayitemcase.md [hunk 23]
```diff
@@ -0,0 +1,30 @@
+---
+slug: /poscreators/middleware-doc/italy/reference-tables/ftpayitemcase
+title: 'Type of Payment: ftPayItemCase'
+---
+
+# Type of Payment: ftPayItemCase
+
+This table expands on the values provided in table [ftPayItemCase in General Part](../../general/reference-tables/reference-tables.md#type-of-payment-ftpayitemcase) with values applicable to the Italian market.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/reference-tables/type-of-payment-ftpayitemcase.md:8]:
```suggestion
This table expands on the values provided in table [ftPayItemCase in General Part](../../general/reference-tables/reference-tables.md#type-of-payment-ftpayitemcase) with values applicable to the italian market.
```

File: doc/middleware-it-registratore-telematico/reference-tables/type-of-receipt-ftreceiptcase.md [hunk 24]
```diff
@@ -0,0 +1,29 @@
+---
+slug: /poscreators/middleware-doc/italy/reference-tables/ftreceiptcase
+title: 'Type of receipt: ftReceiptCase'
+---
+
+# Type of Receipt: ftReceiptCase
+
+The `ftReceiptCase` indicates the receipt type and defines how the fiskaltrust.SecurityMechanism should process it following Italian law.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/reference-tables/type-of-receipt-ftreceiptcase.md:8]:
```suggestion
The `ftReceiptCase` indicates the receipt type and defines how the fiskaltrust.SecurityMechanism should process it following italian law.
```

Comment (volllly) [reply doc/middleware-it-registratore-telematico/reference-tables/type-of-receipt-ftreceiptcase.md:8]: is "fiskaltrust.SecurityMechanism" still a term we use?

File: doc/middleware-it-registratore-telematico/reference-tables/type-of-service-ftchargeitemcase.md [hunk 25]
```diff
@@ -0,0 +1,61 @@
...
+| `0x4954000000000021` | **Account of a third party/ third party name/ collection**<br />For processing, see (`0x4954000000000007`)| TBD |
+| `0x4954000000000022` | **Obligation**                                                                                         | TBD |
+| `0x4954000000000023` | **Discount/Surcharge negativ value is discount, positiv value surcharge: normal Vat 22%**<br /> | TBD |
+| `0x4954000000000024` | **Discount/Surcharge negativ value is discount, positiv value surcharge: reduced-1 Vat 10%**<br /> | TBD |
+| `0x4954000000000025` | **Discount/Surcharge negativ value is discount, positiv value surcharge: reduced-2 Vat  5%**<br /> | TBD |
+| `0x4954000000000026` | **Discount/Surcharge negativ value is discount, positiv value surcharge: special Vat  4%**<br /> | TBD |
+| `0x4954000000000027` | **Discount/Surcharge negativ value is discount, positiv value surcharge: zero Vat  0%**<br /> | TBD |
+
+## ftChargeItemCaseFlag
+This table shows flags that can be added to each `ftChargeItemCase` with values applicable to the Italian market.
```
Comment (volllly) [inline doc/middleware-it-registratore-telematico/reference-tables/type-of-service-ftchargeitemcase.md:55]:
```suggestion
This table shows flags that can be added to each `ftChargeItemCase` with values applicable to the italian market. 
```

File: doc/middleware-it-registratore-telematico/reference-tables/type-of-receipt-ftreceiptcase.md (hunk 24 above)
Comment (mijomilicevic) [reply doc/middleware-it-registratore-telematico/reference-tables/type-of-receipt-ftreceiptcase.md:8]: i think it is. i think what it means is the basic chaining of the receipts we are doing in the queue for every market. but however i am not sure ^^

**Review by Bluette-C-Riviere — APPROVED** [review]

**Review by mijomilicevic — APPROVED** [review]

**Pull request merged**